core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java

```diff
@@ -298,105 +294,25 @@ private boolean isStillReferenced(SubjectInContext key) {
 
 	private synchronized void queueStatement(Statement st) {
 		SubjectInContext key = new SubjectInContext(st);
-		Set<Statement> stmts = statementBySubject.get(key);
+		Set<Statement> stmts = stmtBySubject.get(key);
 		if (stmts == null && st.getSubject() instanceof BNode && !stack.contains(key)) {
 			blanks.add(st);
 		} else {
 			if (stmts == null) {
-				statementBySubject.put(key, stmts = new TreeSet<>(comparator));
+				stmtBySubject.put(key, stmts = new TreeSet<>(comparator));
 			}
 			stmts.add(st);
 		}
 		queueSize++;
 	}
 
 	private synchronized void flushStatements() throws RDFHandlerException {
-		if (!statementBySubject.isEmpty() || !blanks.isEmpty()) {
+		if (!stmtBySubject.isEmpty() || !blanks.isEmpty()) {
 			flushNamespaces();
-
-			// used to store all the statements
-			ArrayList<Statement> statements = new ArrayList<>();
-
-			// used to store blank nodes along with the number of times they are used as an object in a statement.
-			Map<BNode, Integer> bNodeOccurrences = new HashMap<>();
-
 			Statement st;
 			while ((st = nextStatement()) != null) {
-				statements.add(st);
-
-				Value obj = st.getObject();
-
-				// if the object in the statement is a blank node, we will update its number of occurrences
-				if (obj instanceof BNode) {
-					bNodeOccurrences.compute(
-							(BNode) obj,
-							(key, i) -> i == null ? 1 : i + 1
-					);
-
-					if (bNodeOccurrences.get(obj) > 1) {
-						if (st.getSubject() instanceof BNode) {
-							nonInlinedNodes.add(st.getSubject());
-						}
-						nonInlinedNodes.add(obj);
-					}
-				}
+				delegate.handleStatement(st);
```

Comment:
This looks like you've reverted quite a lot of history.

Reply:
It's really only two pull requests - yours and Miro's. In retrospect it wasn't really necessary but I wanted to start with a fresh perspective instead of continuing to build on patches that didn't completely solve the problems.